Geometry_Engine/Query/IsContaining.cs

```diff
@@ -709,6 +709,168 @@ public static bool IsContaining(this Cuboid cuboid, IGeometry geometry, bool acc
+        [Input("acceptOnEdges", "Decide whether to allow the point to sit on the edge of the panel, default false.")]
+        [Input("acceptPartialContainment", "Decide whether to allow some of the points to sit outside the panels as long as at least one is within them.")]
+        [Input("tolerance", "Distance tolerance to use to determine intersections.")]
+        [Output("isContaining", "True if the points are contained within the bounds of the panels, false if they are not.")]
+        public static bool IsContaining(this List<Polyline> closedVolume, List<Point> points, bool acceptOnEdges = false, bool acceptPartialContainment = false, double tolerance = BH.oM.Geometry.Tolerance.Distance)
+        {
+            List<List<Point>> pointLists = points.Select(x => new List<Point>() { x }).ToList();
+            List<bool> ptContained = IsContaining(closedVolume, pointLists, acceptOnEdges, false, tolerance);
+            if (acceptPartialContainment)
+                return ptContained.Any(x => x);
+            else
+                return ptContained.All(x => x);
+        }
+
+        /***************************************************/
+
+        [Description("Checks if a point lies within a collection of polylines describing a closed volume contains each of a provided list of list of points.")]
+        [Input("closedVolume", "A collection of Polylines outlining each of the faces of the closed volume to check with.")]
+        [Input("pointLists", "The List of Lists of points to check to see if each List of points are contained within the bounds of the panels.")]
+        [Input("acceptOnEdges", "Decide whether to allow the points to sit on the edge of the panel, default false.")]
+        [Input("acceptPartialContainment", "Decide whether to allow some of the points to sit outside the panels as long as at least one is within them.")]
+        [Output("isContaining", "True if the points of each sublist are contained within the bounds of the panels, false if it is not for each sublist of points provided.")]
+        public static List<bool> IsContaining(this List<Polyline> closedVolume, List<List<Point>> pointLists, bool acceptOnEdges = false, bool acceptPartialContainment = false, double tolerance = BH.oM.Geometry.Tolerance.Distance)
+        {
+            if (closedVolume == null)
+            {
+                BH.Engine.Base.Compute.RecordError("Cannot query if a collection of polylines contains a point if the polylines are null.");
+                return new List<bool>() { false };
+            }
+
+            List<Point> ctrPoints = new List<Point>();
+            List<Plane> planes = new List<Plane>();
+            foreach (Polyline be in closedVolume)
+            {
+                List<Point> srfPts = be.IControlPoints();
+                Plane fitPlane = srfPts.FitPlane(tolerance);
+                if (fitPlane != null)
+                    planes.Add(fitPlane);
```

Comment:
Does it make sense to run the containment check if a plane is missing? Wouldn't we end up with different collection counts for `closedVolume` and `planes`?

Reply:
Resolved per latest commit from @pawelbaran